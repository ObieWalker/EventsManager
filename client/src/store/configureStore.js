import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import rootReducer from '../reducers/rootReducer';

const middleware = applyMiddleware(thunkMiddleware, createLogger({ collapsed: true }));

const Store = createStore(
    rootReducer,
    applyMiddleware(middleware)
);

export default Store;