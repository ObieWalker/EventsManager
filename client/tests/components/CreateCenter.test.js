import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import connect, { CreateCenter }
  from '../../src/components/AdminDashboard/CreateCenter.jsx';

let mountedComponent;
let props;
const mockStore = configureStore();
const createCenter = jest.fn(() => Promise.resolve({}));

let wrapper;
const createSuccess = '';
const createError = '';
const getComponent = () => {
  if (!mountedComponent) {
    props = {
      createCenter,
      createSuccess: '',
      createError: ''
    };
    mountedComponent = shallow(<CreateCenter {...props} />);
  }
  return mountedComponent;
};

describe('Admin Centers Component', () => {
  beforeEach(() => {});
  wrapper = getComponent();

  it('component successfully rendered', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should set name in state when handleChange is called with name', () => {
    wrapper.instance().handleChange({ target: { name: 'name', value: 'obi' } });
    expect(wrapper.instance().state.name).toEqual('obi');
  });
  it('should have a method that handles focus', () => {
    expect(wrapper.instance().handleOnFocus).toBeDefined();
  });
  it('should have a method that handles handleChange', () => {
    expect(wrapper.instance().handleChange).toBeDefined();
  });
  it('should have a method that handles formIsValid', () => {
    expect(wrapper.instance().formIsValid).toBeDefined();
  });
  it('should have a method that handles formIsValid', () => {
    expect(wrapper.instance().clear).toBeDefined();
  });
  it('testing create center function', () => {
    wrapper = getComponent();
    const e = { preventDefault: () => undefined };
    e.preventDefault = jest.fn();
    wrapper.instance().onSubmit(e);
    expect(e.preventDefault).toHaveBeenCalled();
  });
  it('testing clear function', () => {
    wrapper = getComponent();
    wrapper.instance().clear();
    expect(wrapper.state().name).toEqual('');
    expect(wrapper.state().address).toEqual('');
    expect(wrapper.state().city).toEqual('');
    expect(wrapper.state().capacity).toEqual('');
    expect(wrapper.state().facility).toEqual('');
    expect(wrapper.state().errors).toEqual({});
  });
});

describe('Connected Component', () => {
  describe('Connected mountedLogin', () => {
    it('component successfully rendered', () => {
      const store = mockStore({
        createCenter,
        createSuccess,
        createError
      });
      wrapper = shallow(<connect store={store}
        createCenter={createCenter}
        createSuccess={createSuccess}
        createError={createError}
      />);
      expect(wrapper.length).toBe(1);
    });
  });
});
