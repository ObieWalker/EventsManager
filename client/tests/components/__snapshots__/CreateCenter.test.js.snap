// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Centers Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateCenter
    createCenter={[MockFunction]}
    createError=""
    createSuccess=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="center"
        >
          Add Center Information
        </h4>,
        false,
        <CenterForm
          addressValue=""
          centerNameValue=""
          cityValue=""
          createCenter={[Function]}
          errors={Object {}}
          facilityValue=""
          onChange={[Function]}
          onFocus={[Function]}
          uploadImage={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Center Information",
          "className": "center",
        },
        "ref": null,
        "rendered": "Add Center Information",
        "type": "h4",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addressValue": "",
          "capacityValue": undefined,
          "centerNameValue": "",
          "cityValue": "",
          "createCenter": [Function],
          "errors": Object {},
          "facilityValue": "",
          "onChange": [Function],
          "onFocus": [Function],
          "uploadImage": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="center"
          >
            Add Center Information
          </h4>,
          false,
          <CenterForm
            addressValue=""
            centerNameValue=""
            cityValue=""
            createCenter={[Function]}
            errors={Object {}}
            facilityValue=""
            onChange={[Function]}
            onFocus={[Function]}
            uploadImage={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Center Information",
            "className": "center",
          },
          "ref": null,
          "rendered": "Add Center Information",
          "type": "h4",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addressValue": "",
            "capacityValue": undefined,
            "centerNameValue": "",
            "cityValue": "",
            "createCenter": [Function],
            "errors": Object {},
            "facilityValue": "",
            "onChange": [Function],
            "onFocus": [Function],
            "uploadImage": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
